### PR TITLE
🎁: Form Validation in RC Forms

### PR DESCRIPTION
**Pull request by kjellhaaland**
# Description

- Updates EDS Components to latest available version
- Added Formik form validation
- Restructured components to decouple state and form

Implements equinor/lighthouse-client#1662

### CODE REVIEW

File: src/packages/Admin/src/Components/Modal/CreateStatusModal.tsx [hunk 1]
```diff
@@ -1,85 +1,72 @@
-import { Button, TextField } from '@equinor/eds-core-react-old';
-import { KeyboardEventHandler, useState } from 'react';
+import { Button } from '@equinor/eds-core-react';
+import { Form, FormikValues } from 'formik';
+import * as Yup from 'yup';
```
Comment (MagnusGjosund) [inline src/packages/Admin/src/Components/Modal/CreateStatusModal.tsx:3]: :O

File: src/packages/Admin/src/Components/Modal/CreateStepModal.tsx [hunk 2]
```diff
@@ -1,78 +1,68 @@
-import { Button, TextField } from '@equinor/eds-core-react-old';
-import { KeyboardEventHandler, useState } from 'react';
+import { Button } from '@equinor/eds-core-react';
+import { Form, FormikValues } from 'formik';
+import * as Yup from 'yup';
```
Comment (MagnusGjosund) [inline src/packages/Admin/src/Components/Modal/CreateStepModal.tsx:3]: :O

File: src/packages/Admin/src/Components/Modal/CreateWorkflowModal.tsx [hunk 3]
```diff
@@ -1,17 +1,25 @@
-import { Button, TextField } from '@equinor/eds-core-react-old';
-import { KeyboardEventHandler, useState } from 'react';
+import { Button } from '@equinor/eds-core-react';
+import { Form, FormikValues } from 'formik';
+import * as Yup from 'yup';
```
Comment (MagnusGjosund) [inline src/packages/Admin/src/Components/Modal/CreateWorkflowModal.tsx:3]: :O

File: src/packages/Admin/src/Components/Modal/EditStatusModal.tsx [hunk 4]
```diff
@@ -1,35 +1,43 @@
-import { Button, TextField } from '@equinor/eds-core-react-old';
+import { Button } from '@equinor/eds-core-react';
+import { Form, FormikValues } from 'formik';
+import * as Yup from 'yup';
```
Comment (MagnusGjosund) [inline src/packages/Admin/src/Components/Modal/EditStatusModal.tsx:3]: :O

File: src/packages/Admin/src/Components/Modal/EditWorkflowModal.tsx [hunk 5]
```diff
@@ -1,25 +1,34 @@
-import { Button, TextField } from '@equinor/eds-core-react-old';
+import { Button } from '@equinor/eds-core-react';
+import { Form, FormikValues } from 'formik';
+import * as Yup from 'yup';
```
Comment (MagnusGjosund) [inline src/packages/Admin/src/Components/Modal/EditWorkflowModal.tsx:3]: :O

File: src/packages/Admin/src/Components/Sidesheet/WorkflowStepSidesheet.tsx [hunk 6]
```diff
@@ -1,75 +1,123 @@
...
-            {item.id === '' ? (
-                <WorkflowStepCreateButtonBar actions={actions} />
-            ) : (
-                <WorkflowStepSaveButtonBar actions={actions} />
-            )}
-        </Wrapper>
-    );
-}
+import { Form, FormikValues } from 'formik';
+import * as Yup from 'yup';
```
Comment (MagnusGjosund) [inline src/packages/Admin/src/Components/Sidesheet/WorkflowStepSidesheet.tsx:2]: :O

File: src/packages/EdsForm/src/Components/FormContainer.tsx [hunk 7]
```diff
@@ -0,0 +1,28 @@
+import { Formik, FormikConfig, FormikValues } from 'formik';
+
+type Props = {
+    onCancel?: () => void;
+} & FormikConfig<FormikValues>;
+
+export const FormContainer = (props: Props): JSX.Element => {
+    /*
+    TODO: Handle Key Shortcuts
```
Comment (MagnusGjosund) [inline src/packages/EdsForm/src/Components/FormContainer.tsx:9]: noe som skal bli sett på senere?

File: src/packages/Workflow/src/Components/Modal/SignWithCommentModal.tsx [hunk 8]
```diff
@@ -1,27 +1,35 @@
-import { Button, TextField } from '@equinor/eds-core-react-old';
+import { Button } from '@equinor/eds-core-react';
+import { Form, FormikValues } from 'formik';
+import * as Yup from 'yup';
```
Comment (MagnusGjosund) [inline src/packages/Workflow/src/Components/Modal/SignWithCommentModal.tsx:3]: :O

File: src/packages/Admin/src/Components/Modal/EditStepModal.tsx [hunk 9]
```diff
@@ -1,23 +1,30 @@
-import { Button, TextField } from '@equinor/eds-core-react-old';
+import { Button } from '@equinor/eds-core-react';
+import { Form, FormikValues } from 'formik';
+import * as Yup from 'yup';
```
[the remark this answers is not in the document]
Comment (kjellhaaland) [reply src/packages/Admin/src/Components/Modal/EditStepModal.tsx:3]: Fixed :)

File: src/packages/Admin/src/Components/Modal/CreateWorkflowModal.tsx (hunk 3 above)
Comment (kjellhaaland) [reply src/packages/Admin/src/Components/Modal/CreateWorkflowModal.tsx:3]: Fixed! :)

File: src/packages/Admin/src/Components/Modal/EditWorkflowModal.tsx (hunk 5 above)
Comment (kjellhaaland) [reply src/packages/Admin/src/Components/Modal/EditWorkflowModal.tsx:3]: Fixed! :)

File: src/packages/Admin/src/Components/Sidesheet/WorkflowStepSidesheet.tsx (hunk 6 above)
Comment (kjellhaaland) [reply src/packages/Admin/src/Components/Sidesheet/WorkflowStepSidesheet.tsx:2]: Fixed! :)

File: src/packages/Workflow/src/Components/Modal/SignWithCommentModal.tsx (hunk 8 above)
Comment (kjellhaaland) [reply src/packages/Workflow/src/Components/Modal/SignWithCommentModal.tsx:3]: Fixed! :)

File: src/packages/EdsForm/src/Components/FormContainer.tsx (hunk 7 above)
Comment (kjellhaaland) [reply src/packages/EdsForm/src/Components/FormContainer.tsx:9]: Fixed! :)

File: src/packages/Admin/src/Components/Modal/CreateStepModal.tsx (hunk 2 above)
Comment (kjellhaaland) [reply src/packages/Admin/src/Components/Modal/CreateStepModal.tsx]: Fixed :)

File: src/packages/Admin/src/Components/Modal/EditStatusModal.tsx (hunk 4 above)
Comment (kjellhaaland) [reply src/packages/Admin/src/Components/Modal/EditStatusModal.tsx]: Fixed :)

File: src/packages/Admin/src/Components/Sidesheet/WorkflowStepActions.tsx [hunk 10]
```diff
@@ -0,0 +1,47 @@
+import { Button, Progress } from '@equinor/eds-core-react-old';
```
Comment (Gustav-Eikaas) [inline src/packages/Admin/src/Components/Sidesheet/WorkflowStepActions.tsx:1]: Can you use the new eds package?

Comment (kjellhaaland) [reply src/packages/Admin/src/Components/Sidesheet/WorkflowStepActions.tsx]: Fixed 👍

**Review by Gustav-Eikaas — APPROVED** [review]
Nice workæåø

**Pull request merged**